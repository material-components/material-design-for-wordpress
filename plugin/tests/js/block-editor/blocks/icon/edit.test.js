/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render, screen, fireEvent } from '@testing-library/react';

/**
 * Internal dependencies
 */
import IconEdit from '../../../../../assets/src/block-editor/blocks/icon/edit';

const baseProps = {
	attributes: { iconSize: '24' },
	setAttributes: jest.fn(),
};

const setup = props => {
	return render( <IconEdit { ...props } /> );
};

describe( 'Icon', () => {
	beforeAll( () => {
		global.materialDesign = {
			defaults: {
				colors: {
					primary_color: '#6200ee',
					secondary_color: '#018786',
				},
			},
			customizerUrls: {
				shape: 'http://example.com/shape',
				colors: 'http://example.com/colors',
			},
		};
	} );

	it( 'displays all the panels', () => {
		setup( baseProps );
		expect( screen.getByText( 'Icon' ) ).toBeInTheDocument();
		expect( screen.getByText( 'Colors' ) ).toBeInTheDocument();
		expect( screen.getByText( 'Size' ) ).toBeInTheDocument();
		expect( screen.getByText( 'Search icon' ) ).toBeInTheDocument();
	} );

	it( 'matches snapshot', () => {
		const wrapper = setup( baseProps );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'should display an icon', () => {
		setup( baseProps );

		const container = document.body;
		const matches = container.querySelector( '.material-icons' );

		expect( matches ).toBeInTheDocument();
	} );

	it( 'should display icon with xl size', () => {
		setup( {
			attributes: { iconSize: '48', icon: 'spa' },
			setAttributes: jest.fn(),
		} );

		const container = document.body;
		const matches = container.querySelector( '.material-icons.md-48' );

		expect( matches ).toBeInTheDocument();
	} );

	it( 'should show slider for custom px when `custom` size is selected', async () => {
		const props = {
			attributes: { iconSize: '48' },
			isSelected: true,
			setAttributes: jest.fn(),
		};
		const { container } = setup( props );
		const customButton = container.querySelectorAll(
			'.btn-group__list__list-item:last-child button'
		)[ 0 ];

		await fireEvent.click( customButton );
		expect( props.setAttributes ).toHaveBeenCalledWith( {
			iconSize: 'custom',
		} );
	} );

	it( 'should not display icon picker if icon position is set', () => {
		setup( {
			attributes: {
				iconSize: '48',
			},
			setAttributes: jest.fn(),
		} );

		expect( screen.queryByText( 'Pick custom size(in px)' ) ).toBeNull();
	} );

	it( 'should have center class if align is center', () => {
		setup( { attributes: { align: 'center' }, setAttributes: jest.fn() } );

		const container = document.body;
		const matches = container.querySelector( '.has-text-align-center' );

		expect( matches ).toBeInTheDocument();
	} );
} );
