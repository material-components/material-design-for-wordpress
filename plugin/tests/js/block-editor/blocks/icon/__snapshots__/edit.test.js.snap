// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <p
      class="a11y-speak-intro-text"
      hidden="hidden"
      id="a11y-speak-intro-text"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    >
      Notifications
    </p>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class=""
      >
        <i
          class="material-icons md-24"
        />
      </div>
      <div
        class="components-panel__body is-opened"
      >
        <h2
          class="components-panel__body-title"
        >
          <button
            aria-expanded="true"
            class="components-button components-panel__body-toggle"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              <svg
                aria-hidden="true"
                class="components-panel__arrow"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 8l-6 5.4 1 1.2 5-4.6 5 4.6 1-1.2z"
                />
              </svg>
            </span>
            Icon
          </button>
        </h2>
        <section
          class="icons-search"
        >
          <div
            class="icons-search__search-input"
          >
            <div
              class="components-base-control"
            >
              <div
                class="components-base-control__field"
              >
                <label
                  class="components-base-control__label"
                  for="inspector-text-control-1"
                >
                  Search icon
                </label>
                <input
                  class="components-text-control__input"
                  id="inspector-text-control-1"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </section>
        <section
          class="icons-container"
        >
          <div
            class="icons-container__icon"
          >
            <button
              class="icons-container__icon__icon-btn"
              type="button"
            >
              <i
                class="material-icons"
              >
                airplanemode_active
              </i>
            </button>
          </div>
          <div
            class="icons-container__icon"
          >
            <button
              class="icons-container__icon__icon-btn"
              type="button"
            >
              <i
                class="material-icons"
              >
                airplanemode_inactive
              </i>
            </button>
          </div>
          <div
            class="icons-container__icon"
          >
            <button
              class="icons-container__icon__icon-btn"
              type="button"
            >
              <i
                class="material-icons"
              >
                bookmark_border
              </i>
            </button>
          </div>
          <div
            class="icons-container__icon"
          >
            <button
              class="icons-container__icon__icon-btn"
              type="button"
            >
              <i
                class="material-icons"
              >
                camera_enhance
              </i>
            </button>
          </div>
          <div
            class="icons-container__icon"
          >
            <button
              class="icons-container__icon__icon-btn"
              type="button"
            >
              <i
                class="material-icons"
              >
                card_giftcard
              </i>
            </button>
          </div>
          <div
            class="icons-container__icon"
          >
            <button
              class="icons-container__icon__icon-btn"
              type="button"
            >
              <i
                class="material-icons"
              >
                card_membership
              </i>
            </button>
          </div>
          <div
            class="icons-container__icon"
          >
            <button
              class="icons-container__icon__icon-btn"
              type="button"
            >
              <i
                class="material-icons"
              >
                card_travel
              </i>
            </button>
          </div>
          <div
            class="icons-container__icon"
          >
            <button
              class="icons-container__icon__icon-btn"
              type="button"
            >
              <i
                class="material-icons"
              >
                chat_bubble
              </i>
            </button>
          </div>
        </section>
      </div>
      <div
        class="components-panel__body is-opened"
      >
        <h2
          class="components-panel__body-title"
        >
          <button
            aria-expanded="true"
            class="components-button components-panel__body-toggle"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              <svg
                aria-hidden="true"
                class="components-panel__arrow"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 8l-6 5.4 1 1.2 5-4.6 5 4.6 1-1.2z"
                />
              </svg>
            </span>
            Colors
          </button>
        </h2>
        <div
          class="components-base-control"
        >
          Overrides will only apply to this Icon. Change Primary Color in 
          <a
            href="http://example.com/colors"
            rel="noreferrer noopener"
            target="_blank"
          >
            Material Design Options
          </a>
           to update all Icons.
        </div>
        <div
          class="components-base-control material-component-color-palette"
        >
          <div
            class="components-base-control__field"
          >
            <span
              class="components-base-control__label"
            >
              <span
                class="material-component-color-palette__label"
              >
                Icon Color
              </span>
              <span
                aria-label="Icon Color"
                class="component-color-indicator"
                style="background: rgb(98, 0, 238);"
              />
            </span>
            <div
              class="material-colors-wrap"
            >
              <div
                class="components-circular-option-picker"
              >
                <div
                  class="components-circular-option-picker__option-wrapper"
                >
                  <button
                    aria-label="Color: Primary"
                    class="components-button components-circular-option-picker__option is-pressed"
                    style="background-color: rgb(98, 0, 238); color: rgb(98, 0, 238);"
                    type="button"
                  />
                  <svg
                    aria-hidden="true"
                    class="dashicon dashicons-saved"
                    focusable="false"
                    height="20"
                    role="img"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15.3 5.3l-6.8 6.8-2.8-2.8-1.4 1.4 4.2 4.2 8.2-8.2"
                    />
                  </svg>
                </div>
                <div
                  class="components-circular-option-picker__option-wrapper"
                >
                  <button
                    aria-label="Color: Secondary"
                    class="components-button components-circular-option-picker__option"
                    style="background-color: rgb(1, 135, 134); color: rgb(1, 135, 134);"
                    type="button"
                  />
                </div>
                <div
                  class="components-circular-option-picker__custom-clear-wrapper"
                >
                  <div
                    class="components-dropdown components-circular-option-picker__dropdown-link-action components-material-color-palette__link"
                  >
                    <button
                      aria-expanded="false"
                      aria-label="Color palette"
                      class="components-button is-link"
                      type="button"
                    >
                      Color palette
                    </button>
                  </div>
                  <div
                    class="components-dropdown components-circular-option-picker__dropdown-link-action"
                  >
                    <button
                      aria-expanded="false"
                      aria-label="Custom color picker"
                      class="components-button is-link"
                      type="button"
                    >
                      Custom Color
                    </button>
                  </div>
                  <button
                    class="components-button components-circular-option-picker__clear is-small"
                    type="button"
                  >
                    Reset
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="components-panel__body is-opened"
      >
        <h2
          class="components-panel__body-title"
        >
          <button
            aria-expanded="true"
            class="components-button components-panel__body-toggle"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              <svg
                aria-hidden="true"
                class="components-panel__arrow"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 8l-6 5.4 1 1.2 5-4.6 5 4.6 1-1.2z"
                />
              </svg>
            </span>
            Size
          </button>
        </h2>
        <div
          class="btn-group"
        >
          <ul
            class="btn-group__list"
          >
            <li
              class="btn-group__list__list-item"
            >
              <button
                class="btn-group__list__list-item__button"
                type="button"
              >
                Small
              </button>
            </li>
            <li
              class="btn-group__list__list-item"
            >
              <button
                class="btn-group__list__list-item__button btn-group__list__list-item__button--active"
                type="button"
              >
                Medium
              </button>
            </li>
            <li
              class="btn-group__list__list-item"
            >
              <button
                class="btn-group__list__list-item__button"
                type="button"
              >
                Large
              </button>
            </li>
            <li
              class="btn-group__list__list-item"
            >
              <button
                class="btn-group__list__list-item__button"
                type="button"
              >
                XL
              </button>
            </li>
            <li
              class="btn-group__list__list-item"
            >
              <button
                class="btn-group__list__list-item__button"
                type="button"
              >
                Custom
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=""
    >
      <i
        class="material-icons md-24"
      />
    </div>
    <div
      class="components-panel__body is-opened"
    >
      <h2
        class="components-panel__body-title"
      >
        <button
          aria-expanded="true"
          class="components-button components-panel__body-toggle"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            <svg
              aria-hidden="true"
              class="components-panel__arrow"
              focusable="false"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 8l-6 5.4 1 1.2 5-4.6 5 4.6 1-1.2z"
              />
            </svg>
          </span>
          Icon
        </button>
      </h2>
      <section
        class="icons-search"
      >
        <div
          class="icons-search__search-input"
        >
          <div
            class="components-base-control"
          >
            <div
              class="components-base-control__field"
            >
              <label
                class="components-base-control__label"
                for="inspector-text-control-1"
              >
                Search icon
              </label>
              <input
                class="components-text-control__input"
                id="inspector-text-control-1"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </section>
      <section
        class="icons-container"
      >
        <div
          class="icons-container__icon"
        >
          <button
            class="icons-container__icon__icon-btn"
            type="button"
          >
            <i
              class="material-icons"
            >
              airplanemode_active
            </i>
          </button>
        </div>
        <div
          class="icons-container__icon"
        >
          <button
            class="icons-container__icon__icon-btn"
            type="button"
          >
            <i
              class="material-icons"
            >
              airplanemode_inactive
            </i>
          </button>
        </div>
        <div
          class="icons-container__icon"
        >
          <button
            class="icons-container__icon__icon-btn"
            type="button"
          >
            <i
              class="material-icons"
            >
              bookmark_border
            </i>
          </button>
        </div>
        <div
          class="icons-container__icon"
        >
          <button
            class="icons-container__icon__icon-btn"
            type="button"
          >
            <i
              class="material-icons"
            >
              camera_enhance
            </i>
          </button>
        </div>
        <div
          class="icons-container__icon"
        >
          <button
            class="icons-container__icon__icon-btn"
            type="button"
          >
            <i
              class="material-icons"
            >
              card_giftcard
            </i>
          </button>
        </div>
        <div
          class="icons-container__icon"
        >
          <button
            class="icons-container__icon__icon-btn"
            type="button"
          >
            <i
              class="material-icons"
            >
              card_membership
            </i>
          </button>
        </div>
        <div
          class="icons-container__icon"
        >
          <button
            class="icons-container__icon__icon-btn"
            type="button"
          >
            <i
              class="material-icons"
            >
              card_travel
            </i>
          </button>
        </div>
        <div
          class="icons-container__icon"
        >
          <button
            class="icons-container__icon__icon-btn"
            type="button"
          >
            <i
              class="material-icons"
            >
              chat_bubble
            </i>
          </button>
        </div>
      </section>
    </div>
    <div
      class="components-panel__body is-opened"
    >
      <h2
        class="components-panel__body-title"
      >
        <button
          aria-expanded="true"
          class="components-button components-panel__body-toggle"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            <svg
              aria-hidden="true"
              class="components-panel__arrow"
              focusable="false"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 8l-6 5.4 1 1.2 5-4.6 5 4.6 1-1.2z"
              />
            </svg>
          </span>
          Colors
        </button>
      </h2>
      <div
        class="components-base-control"
      >
        Overrides will only apply to this Icon. Change Primary Color in 
        <a
          href="http://example.com/colors"
          rel="noreferrer noopener"
          target="_blank"
        >
          Material Design Options
        </a>
         to update all Icons.
      </div>
      <div
        class="components-base-control material-component-color-palette"
      >
        <div
          class="components-base-control__field"
        >
          <span
            class="components-base-control__label"
          >
            <span
              class="material-component-color-palette__label"
            >
              Icon Color
            </span>
            <span
              aria-label="Icon Color"
              class="component-color-indicator"
              style="background: rgb(98, 0, 238);"
            />
          </span>
          <div
            class="material-colors-wrap"
          >
            <div
              class="components-circular-option-picker"
            >
              <div
                class="components-circular-option-picker__option-wrapper"
              >
                <button
                  aria-label="Color: Primary"
                  class="components-button components-circular-option-picker__option is-pressed"
                  style="background-color: rgb(98, 0, 238); color: rgb(98, 0, 238);"
                  type="button"
                />
                <svg
                  aria-hidden="true"
                  class="dashicon dashicons-saved"
                  focusable="false"
                  height="20"
                  role="img"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M15.3 5.3l-6.8 6.8-2.8-2.8-1.4 1.4 4.2 4.2 8.2-8.2"
                  />
                </svg>
              </div>
              <div
                class="components-circular-option-picker__option-wrapper"
              >
                <button
                  aria-label="Color: Secondary"
                  class="components-button components-circular-option-picker__option"
                  style="background-color: rgb(1, 135, 134); color: rgb(1, 135, 134);"
                  type="button"
                />
              </div>
              <div
                class="components-circular-option-picker__custom-clear-wrapper"
              >
                <div
                  class="components-dropdown components-circular-option-picker__dropdown-link-action components-material-color-palette__link"
                >
                  <button
                    aria-expanded="false"
                    aria-label="Color palette"
                    class="components-button is-link"
                    type="button"
                  >
                    Color palette
                  </button>
                </div>
                <div
                  class="components-dropdown components-circular-option-picker__dropdown-link-action"
                >
                  <button
                    aria-expanded="false"
                    aria-label="Custom color picker"
                    class="components-button is-link"
                    type="button"
                  >
                    Custom Color
                  </button>
                </div>
                <button
                  class="components-button components-circular-option-picker__clear is-small"
                  type="button"
                >
                  Reset
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="components-panel__body is-opened"
    >
      <h2
        class="components-panel__body-title"
      >
        <button
          aria-expanded="true"
          class="components-button components-panel__body-toggle"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            <svg
              aria-hidden="true"
              class="components-panel__arrow"
              focusable="false"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 8l-6 5.4 1 1.2 5-4.6 5 4.6 1-1.2z"
              />
            </svg>
          </span>
          Size
        </button>
      </h2>
      <div
        class="btn-group"
      >
        <ul
          class="btn-group__list"
        >
          <li
            class="btn-group__list__list-item"
          >
            <button
              class="btn-group__list__list-item__button"
              type="button"
            >
              Small
            </button>
          </li>
          <li
            class="btn-group__list__list-item"
          >
            <button
              class="btn-group__list__list-item__button btn-group__list__list-item__button--active"
              type="button"
            >
              Medium
            </button>
          </li>
          <li
            class="btn-group__list__list-item"
          >
            <button
              class="btn-group__list__list-item__button"
              type="button"
            >
              Large
            </button>
          </li>
          <li
            class="btn-group__list__list-item"
          >
            <button
              class="btn-group__list__list-item__button"
              type="button"
            >
              XL
            </button>
          </li>
          <li
            class="btn-group__list__list-item"
          >
            <button
              class="btn-group__list__list-item__button"
              type="button"
            >
              Custom
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
